### ISSUE TITLE
💡 RFC: Respect fetch cache-control headers

### ISSUE THREAD

**Issue by FredKSchott** [withastro/astro#693]
### Motivation

- Spun out from this RFC: https://github.com/snowpackjs/astro/issues/619
- Its common to use `fetch()` in an Astro file, which may get rerun multiple times when rendering a page (for example, if many of the same components exist on the same page). This creates a perf foot-gun for users.

### Proposed Solution

- We can reduce the cost of fetch() by respecting the cache-control headers of the response.
- For example: If an API response comes back with `Cache-Control: immutable` we are being told that the response won't change over time. We can safely cache this and reuse the cached value the next time fetch is called.

### Open Questions

- Should we respect cache-control headers during dev? What if you're working on an API in parallel with your site, and the API keeps changing?

**Comment by natemoo-re** [comment]
Would we need to roll this ourselves? I've used [`http-cache-semantics`](https://github.com/kornelski/http-cache-semantics) before but I'm hopeful that [`undici-fetch`](https://github.com/Ethan-Arrowood/undici-fetch#readme) has support for this built-in.

**Comment by FredKSchott** [comment]
Hey everyone! Our current RFC process is beginning to break down at this size, with over 50 open RFCs currently in the "discussing" stage. A growing community is a great problem to have, but our ability to give you RFC feedback has suffered as a result. In an effort to improve our RFC process, we are making some changes to better organize things. 

**From now on, all RFCs will live in a standalone repo:** https://github.com/withastro/rfcs

This allows us to do three things: 1) Use threaded discussions for high-level ideas and improvements, without necessarily requiring an implementation for every idea. 2) Improve the quality of our RFC template and the speed/quality of all feedback. 3) Support inline comments and explicit approvals on RFCs, via a new Pull Request review process.

We hope that this new process leads to better RFC weekly calls and faster feedback on your RFCs from maintainers. More detail can be found in the new RFC repo [README](https://github.com/withastro/rfcs).

--- 

**We can't automatically convert this issue to an RFC in the new repo** because new RFC template is more detailed that this one. But, you can still continue this discussion in the new repo by [creating a new Discussion](https://github.com/withastro/rfcs/discussions) in the RFC repo and copy-and-pasting this post (and any relevant follow-up comments) into it. Discussions are available for high-level ideas and suggestions without the requirement of a full implementation proposal.

Then, when you are ready to propose (or re-propose) an implementation for feedback and approval, you can create a new RFC using the new RFC template. More detail about how to do this can be found in the new RFC repo [README](https://github.com/withastro/rfcs).

Thanks for your patience as we attempt to improve things for both authors and reviewers. If you have any questions, don't hesitate to reach out on Discord. https://astro.build/chat
